blockchain/state/state_object.go

```diff
@@ -287,13 +331,25 @@ func (self *stateObject) updateStorageTrie(db Database) Trie {
 		}
 		self.originStorage[key] = value
 
+		var v []byte
 		if (value == common.Hash{}) {
 			self.setError(tr.TryDelete(key[:]))
-			continue
+		} else {
+			// Encoding []byte cannot fail, ok to ignore the error.
+			v, _ = rlp.EncodeToBytes(bytes.TrimLeft(value[:], "\x00"))
+			self.setError(tr.TryUpdate(key[:], v))
+		}
+		// If state snapshotting is active, cache the data til commit
+		if self.db.snap != nil {
+			if storage == nil {
```

Comment:
`storage` is always nil because it's local vaiable. Isn't it?

Reply:
Not always `nil`. Please take a look line 346 and 347.